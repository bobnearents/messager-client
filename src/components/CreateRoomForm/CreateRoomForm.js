import React, { Component } from 'react';
import ApiService from "../../services/api-service";

class CreateRoomForm extends Component {
  constructor(props) {
    super(props);
    this.state = {}
  }

  handleSubmit = ev => {
    ev.preventDefault();
     const { name } = ev.target;

    ApiService.createRoom(
      name.value,
    );
    
    name.value = '';
    this.props.onCreateRoomSuccess();

  }

  render() {
    return (
      <div className="signup-form">
        <h2 className="form-header">Make a Room!</h2>
        <form onSubmit={ev => this.handleSubmit(ev)}>

          <label htmlFor="name">Room Name: </label>
          <input type="text" id="name" />
          <button type="submit">Submit</button>
        </form>
      </div>
    );
  }
}

export default CreateRoomForm;