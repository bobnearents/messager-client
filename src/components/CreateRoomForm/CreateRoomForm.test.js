import React from 'react';
import { shallow } from 'enzyme';
import renderer from "react-test-renderer";


import CreateRoomForm from './CreateRoomForm';

describe('<CreateRoomForm />', () => {

  it('Renders without crashing', () => {
    shallow(<CreateRoomForm />)
  })

  it("renders correctly", () => {
    const tree = renderer.create(<CreateRoomForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //implement mocking
  //https://medium.com/@rishabhsrao/mocking-and-testing-fetch-with-jest-c4d670e2e167

})