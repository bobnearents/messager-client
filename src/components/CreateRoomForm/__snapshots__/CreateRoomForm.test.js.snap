// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRoomForm /> renders correctly 1`] = `
<div
  className="signup-form"
>
  <h2
    className="form-header"
  >
    Make a Room!
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Room Name: 
    </label>
    <input
      id="name"
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
